tests/sentry/api/endpoints/notifications/test_notification_actions_index.py

```diff
@@ -36,18 +40,32 @@ def setUp(self):
             self.create_project(name="greenpath", organization=self.organization),
             self.create_project(name="dirtmouth", organization=self.organization),
         ]
-        self.base_data = {
+        self.base_data: MutableMapping[str, Any] = {
             "serviceType": "email",
             "triggerType": "audit-log",
             "targetType": "specific",
             "targetDisplay": "@hollowknight",
             "targetIdentifier": "THK",
         }
-        self.mock_register = lambda data: NotificationAction.register_action(
-            trigger_type=ActionTrigger.get_value(data["triggerType"]),
-            service_type=ActionService.get_value(data["serviceType"]),
-            target_type=ActionTarget.get_value(data["targetType"]),
-        )
+
+        def mock_register(
+            data: MutableMapping[str, Any]
+        ) -> Callable[[type[ActionRegistrationT]], type[ActionRegistrationT]]:
+            assert (
+                triggerType := ActionTrigger.get_value(data["triggerType"])
+            ) is not None, "triggerType must exist"
```

Comment:
I tend to recommend against assignment expressions -- it makes this much harder to read than just:

```
trigger_type = ...
assert trigger_type is not None
```

and a lot easier to not mess up the parens (if they're in the wrong spot you're just assigning a bunch of bools accidentally!)